src/libopensc/sc.c

```diff
@@ -76,11 +77,13 @@ int sc_hex_to_bin(const char *in, u8 *out, size_t *outlen)
 			}
 			byte |= c;
 		}
-		// Detect premature end of string before byte is complete
-		if (!*in && nybbles >= 0) {
+
+		/* Detect premature end of string before byte is complete */
+		if (in_len > 1 && *in == '\0' && nybbles >= 0) {
```

Comment:
You still want to allow `in_len == 0`, though we eliminated the use of single digit hex strings?